src/class-convertkit-api.php

```diff
@@ -2232,4 +2232,27 @@ private function get_error_message( $key ) {
 
 	}
 
+	/**
+	 * Helper method to mask all but the last 4 characters of a string.
+	 *
+	 * @since   1.4.2
+	 *
+	 * @param   string $str    String to mask.
+	 * @return  string          Masked string
+	 */
+	private function mask_string( $str ) {
+
+		// Don't mask if less than 4 characters.
+		if ( strlen( $str ) < 4 ) {
+			return $str;
+		}
+
+		return str_replace(
```

Comment:
The most minor of nits: I don't see why it's necessary to use str_replace when we could just put `str_repeat( '*', ( strlen( $str ) - 4 ) ) . substr( $str, -4 )`. I also think it's fine if we just use a single asterisk or three like an ellipsis, such as when credit card numbers are masked.

Reply:
> The most minor of nits: I don't see why it's necessary to use str_replace when we could just put `str_repeat( '*', ( strlen( $str ) - 4 ) ) . substr( $str, -4 )`

Agreed

> I also think it's fine if we just use a single asterisk or three like an ellipsis, such as when credit card numbers are masked.

I'd prefer to retain the length of the original string at this time, as we're masking most of the signed subscriber ID, first name and email address. For logs provided as part of support, it then makes it a little easier to e.g. identify an email address within a ConvertKit account to confirm it subscribed. Otherwise, the logging itself is probably not that helpful.